examples/demo/server/main.go

```diff
@@ -0,0 +1,171 @@
+
+// Initializes an OTLP exporter, and configures the corresponding trace and
+// metric providers.
+func initProvider() func() {
+	ctx := context.Background()
+
+	otelAgentAddr, ok := os.LookupEnv("OTEL_AGENT_ENDPOINT")
+	if !ok {
+		otelAgentAddr = "0.0.0.0:4317"
+	}
+
+	metricClient := otlpmetricgrpc.NewClient(
+		otlpmetricgrpc.WithInsecure(),
+		otlpmetricgrpc.WithEndpoint(otelAgentAddr))
+	metricExp, err := otlpmetric.New(ctx, metricClient)
+	handleErr(err, "Failed to create the collector metric exporter")
+
+	pusher := controller.New(
+		processor.New(
+			simple.NewWithExactDistribution(),
+			metricExp,
+		),
+		controller.WithExporter(metricExp),
+		controller.WithCollectPeriod(2*time.Second),
+	)
+	global.SetMeterProvider(pusher.MeterProvider())
+
+	err = pusher.Start(ctx)
+	handleErr(err, "Failed to start metric pusher")
+
+	traceClient := otlptracegrpc.NewClient(
+		otlptracegrpc.WithInsecure(),
+		otlptracegrpc.WithEndpoint(otelAgentAddr),
+		otlptracegrpc.WithDialOption(grpc.WithBlock()))
+	traceExp, err := otlptrace.New(ctx, traceClient)
+	handleErr(err, "Failed to create the collector trace exporter")
+
+	res, err := resource.New(ctx,
+		resource.WithFromEnv(),
+		resource.WithProcess(),
+		resource.WithTelemetrySDK(),
+		resource.WithHost(),
+		resource.WithAttributes(
+			// the service name used to display traces in backends
+			semconv.ServiceNameKey.String("demo-server"),
+		),
+	)
+	handleErr(err, "failed to create resource")
+
+	bsp := sdktrace.NewBatchSpanProcessor(traceExp)
+	tracerProvider := sdktrace.NewTracerProvider(
+		sdktrace.WithSampler(sdktrace.AlwaysSample()),
+		sdktrace.WithResource(res),
+		sdktrace.WithSpanProcessor(bsp),
+	)
+
+	// set global propagator to tracecontext (the default is no-op).
+	otel.SetTextMapPropagator(propagation.TraceContext{})
+	otel.SetTracerProvider(tracerProvider)
+
+	return func() {
+		cxt, cancel := context.WithTimeout(ctx, time.Second)
+		defer cancel()
+		if err := traceExp.Shutdown(cxt); err != nil {
+			otel.Handle(err)
+		}
+		// pushes any last exports to the receiver
+		if err := pusher.Stop(cxt); err != nil {
+			otel.Handle(err)
+		}
+	}
+}
+
+func handleErr(err error, message string) {
+	if err != nil {
+		log.Fatalf("%s: %v", message, err)
+	}
+}
+
+func main() {
+	shutdown := initProvider()
+	defer shutdown()
+
+	meter := global.Meter("demo-server-meter")
+	serverAttribute := attribute.String("server-attribute", "foo")
+	commonLabels := []attribute.KeyValue{serverAttribute}
+	requestCount := metric.Must(meter).NewInt64Counter(
+		"demo_server/request_counts",
+		metric.WithDescription("The number of requests received"),
+	)
+
+	// create a handler wrapped in OpenTelemetry instrumentation
+	handler := http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		//  random sleep to simulate latency
+		var sleep int64
+		switch modulus := time.Now().Unix() % 5; modulus {
+		case 0:
+			sleep = rng.Int63n(17001)
```

Comment:
17001 ms seems like potentially a high number for simulated latency 😄

Reply:
Good point :)